source/geometry_model/chunk.cc

```diff
@@ -249,6 +249,47 @@ namespace aspect
 
 
 
+      template <int dim>
+      Tensor<1, dim>
+      ChunkGeometry<dim>::
+      normal_vector(const typename Triangulation<dim>::face_iterator &face,
+                    const Point<dim> &p) const
+      {
+        // if the maximum deviation for the distance from the vertices to the center
+        // is less than 1.e-5 of the minimum distance to the first vertex, assume we
+        // can simply return p-center. otherwise, we compute the normal using
+        // get_normal_vector
+        constexpr unsigned int n_vertices = GeometryInfo<dim>::vertices_per_face;
```

Comment:
If you don't have this function in place, what function would be called instead? The one that computes the normal based on pull-backs and push-forwards?

The issue in question is not that of the computation of the normal vector, but what the point `p` really is. In essence, we are given a point `p` in real coordinates of the cell. We need to find an accurate representation of the point in the reference coordinate system to use the pull-back/push-forward machinery. I think that what you are doing here makes sense to me -- it just computes a more accurate representation of the normal vector.

You might want to copy the documentation improvements from https://github.com/dealii/dealii/pull/9559.